ShopNow/UserApp/views.py

```diff
@@ -30,8 +30,21 @@
 from django.core.exceptions import ValidationError
 from django.db.utils import IntegrityError
 
+from Cart.models import cartModel
+
+from django.views import View
+import requests
+
+# from order.models import order
+# from order.views import orderApi
+
+
+# to auhtenticatet the user after succesfull logina and setting the tokens in sessions,
+from django.contrib.auth import authenticate, login
+
+
+
 
-@csrf_exempt
 @api_view(['POST'])
 def registerUser(request):
     if request.method == 'POST':
```

Comment:
Since the function is already restricted to POST via the @api_view(['POST']) decorator, there’s no need to check the request method explicitly like if request.method == 'POST': it is a redundant check so that we can remove this.

Please find this here and all other places in this view also.